public/components/trace_analytics/components/common/plots/service_map.tsx

```diff
@@ -163,6 +185,23 @@ export function ServiceMap({
     autoResize: true,
   };
 
+  const setZoomLimits = (networkInstance) => {
+    let lastZoomLevel = 1.0;
+    const initialPosition = networkInstance.getViewPosition();
+
+    networkInstance.on('zoom', (params) => {
+      const zoomLevel = params.scale;
+
+      if (zoomLevel < 0.25 && zoomLevel < lastZoomLevel) {
+        networkInstance.moveTo({ scale: 0.25, position: initialPosition });
+      } else if (zoomLevel > 1.75) {
+        networkInstance.moveTo({ scale: 1.75 });
+      }
```

Comment:
Just to get an idea of whats happening here, we're adding a boundary to the scale of the screen so that if these limits are reached, it'll change the scale back, but what does `zoomLevel < lastZoomLevel` do here and why isn't it included in the conditional when we go too big? My first assumption was that it was to allow for the screen to gradually zoom back out, but why wouldn't we want it for `zoomLevel > 1.75`?
Also, why is position set in one condition and not the other, what does it do?

Reply:
Because there are no zoom limits available to bet set we are handling this using the moveTo function. This functionality sets a limit on if the user zooms in to much if they reach the limit it stops it, as a side effect there is a little movement if the user continuously zooms in while at the limit. 
The zoom out limit has the added functionality to reset the node cluster to the center to prevent the user from getting lost. This is only present on the zoom out as we would not want to recenter and lose position if the user accidentally zoomed in to much.

Video example of behavior at limits:

https://github.com/user-attachments/assets/df088c21-196a-4292-a144-207ae5037f2a

Reply:
ah that makes sense thanks